### DISCUSSION TITLE
Question regarding submitting draft proposals

### DISCUSSION THREAD

**Discussion by abdelrahmanmagdii** [cncf/mentoring#628] [Q&A] [answered]
Hello CNCF!

I was wondering how should I submit a draft proposal in order to possibly get feedback before submitting my final one? I believe it might be through the GSoC website's submit proposal option, but I am afraid it might be counted as the final draft and therefore I thought I should ask before.

Also, I was wondering if anyone could point me in the direction of any Argo CD extension documentation so I can further investigate how I can implement it to Argo Workflow.

Thank you in advance!
Abdelrahman Amer

**Comment by nate-double-u** [comment, marked as answer]
Hi @abdelrahmanmagdii!

I think your question is similar to this one in the FAQ:
[Should I send proposals directly to the mentoring organizations?](https://developers.google.com/open-source/gsoc/faq#should_i_send_proposals_directly_to_the_mentoring_organizations)

Proposals need to be submitted through the program site. Reviews, however, are done with the project mentors in the manner they prefer. I recommend reaching out and talking with the project mentors about any project you want to submit.
[
Proposals can be edited as many times as you like](https://google.github.io/gsocguides/student/writing-a-proposal#submit-a-proposal-early) before the deadline (April 19 - 18:00 UTC)

As for Argo specific documentation, I'd suggest starting with [this discussion, #565](https://github.com/cncf/mentoring/discussions/565), and reaching out to the Argo maintainers.

**Reply by abdelrahmanmagdii** [reply]
Hello @nate-double-u,

Thank you very much for your reply! Ill try contacting my proposed idea's respective mentors.

Best regards,
Abdelrahman
